scaleway/resource_baremetal_server_beta.go

```diff
@@ -0,0 +1,220 @@
+package scaleway
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	baremetal "github.com/scaleway/scaleway-sdk-go/api/baremetal/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayBaremetalServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayBaremetalServerCreate,
+		Read:   resourceScalewayBaremetalServerRead,
+		Update: resourceScalewayBaremetalServerUpdate,
+		Delete: resourceScalewayBaremetalServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Name of the server.",
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "ID of the server type.",
+				ValidateFunc: validationUUID(),
+			},
+			"image_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "The base image of the server", // TODO: add in doc example with UUID
+				ValidateFunc: validationUUID(),
+			},
+			"ssh_key_ids": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validationUUID(),
+				},
+				Required:    true,
+				ForceNew:    true,
```

Comment:
🤔 If i add/remove/update an ssh key this will wipe out my server ?

Reply:
yes